lib/components/mobile_overlay_menu/mobile_overlay_menu.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:fluentui_system_icons/fluentui_system_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:tarka_ui/components/menu_item/menu_item.dart';
+import 'package:tarka_ui/components/mobile_overlay_footer/mobile_overlay_footer.dart';
+import 'package:tarka_ui/components/mobile_overlay_header/mobile_overlay_header.dart';
+
+import '../../styles/theme.dart';
+
+class TUIMobileOverlayMenu extends StatelessWidget {
+  final String title;
+  final VoidCallback? action;
+  final double padding = 16;
+  final double verticalGap = 8;
+  late List<MenuItem> menuItems;
+
+  TUIMobileOverlayMenu({
+    super.key,
+    required this.title,
+    required this.action,
+    required List<TUIMenuItem> menuItems,
+  }) {
+    final menuItemsList = <MenuItem>[];
+
+    for (int index = 0; index < menuItems.length; index++) {
+      final menuItem = MenuItem(
+        view: menuItems[index],
+        index: index,
+      );
+      menuItemsList.add(menuItem);
+    }
+
+    this.menuItems = menuItemsList;
```

Comment:
Any reason for wrapping `List<TUIMenuItem>` inside `List<MenuItem>`?

Reply:
As I have mentioned in other comment, on iOS the index value is used for accessibility MenuItem is just a wrapper to add the index value.